From fb6d069db28a4abb165f8d48319676b07c3c3491 Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Wed, 17 Sep 2014 19:53:01 +0200
Subject: [PATCH] udev: rules - close empty file

If the file is found to be empty, we exit early without closing the file first.

Found by coverity. Fixes CID #1237759.

(cherry picked from commit 6c8aaf0c1bf2a04010ae7125a08ceb51e7058712)

Conflicts:
	src/udev/udev-rules.c
---
 src/udev/udev-rules.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/udev/udev-rules.c b/src/udev/udev-rules.c
index 7e6db812c9..3b7e263ab2 100644
--- a/src/udev/udev-rules.c
+++ b/src/udev/udev-rules.c
@@ -1514,9 +1514,8 @@ invalid:
         return -1;
 }
 
-static int parse_file(struct udev_rules *rules, const char *filename)
-{
-        FILE *f;
+static int parse_file(struct udev_rules *rules, const char *filename) {
+        _cleanup_fclose_ FILE *f = NULL;
         unsigned int first_token;
         unsigned int filename_off;
         char line[UTIL_LINE_SIZE];
@@ -1570,7 +1569,6 @@ static int parse_file(struct udev_rules *rules, const char *filename)
                 }
                 add_rule(rules, key, filename, filename_off, line_nr);
         }
-        fclose(f);
 
         /* link GOTOs to LABEL rules in this file to be able to fast-forward */
         for (i = first_token+1; i < rules->token_cur; i++) {
